### PR TITLE
support to generate xxTypeRefer for scala enum

### PR DESCRIPTION
**Pull request by jxnu-liguobin**
By the way, extract the version number of the test project and change readme to one example.
https://github.com/kobylynskyi/graphql-java-codegen/issues/479

### CODE REVIEW

File: src/main/resources/templates/scala-lang/scalaClassGraphqlEnum.ftl [hunk 1]
```diff
@@ -2,6 +2,8 @@
 package ${package}
 
 </#if>
+import com.fasterxml.jackson.core.`type`.TypeReference
```
Comment (kobylynskyi) [inline src/main/resources/templates/scala-lang/scalaClassGraphqlEnum.ftl:5]: @jxnu-liguobin, do you need to wrap this import into `<#if serializationLibrary == 'JACKSON'>`?

**Review by kobylynskyi — CHANGES_REQUESTED** [review]

File: src/main/resources/templates/scala-lang/scalaClassGraphqlEnum.ftl (hunk 1 above)
Comment (jxnu-liguobin) [reply src/main/resources/templates/scala-lang/scalaClassGraphqlEnum.ftl:5]: yes, will change

File: src/test/resources/expected-classes/scala/deprecated/Status.scala.txt [hunk 2]
```diff
@@ -1,5 +1,7 @@
 package com.github.graphql
 
+import com.fasterxml.jackson.core.`type`.TypeReference
```
Comment (kobylynskyi) [inline src/test/resources/expected-classes/scala/deprecated/Status.scala.txt:3]: Do you also need to do the “if JACKSON” check?

**Review by kobylynskyi — CHANGES_REQUESTED** [review]

File: src/test/resources/expected-classes/scala/deprecated/Status.scala.txt (hunk 2 above)
Comment (jxnu-liguobin) [reply src/test/resources/expected-classes/scala/deprecated/Status.scala.txt:3]: Only enumeration is needed, and only when Jackson is used

**Pull request merged**